src/Time/StepChoosers/RelaxTimeStepper.hpp

```diff
@@ -0,0 +1,84 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cmath>
+#include <limits>
+#include <pup.h>
+
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Time/StepChoosers/StepChooser.hpp"  // IWYU pragma: keep
+#include "Utilities/Registration.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class Time;
+namespace Parallel {
+template <typename Metavariables>
+class ConstGlobalCache;
+}  // namespace Parallel
+namespace Tags {
+struct HistoryEvolvedVariables;
+struct SubstepTime;
+}  // namespace Tags
+/// \endcond
+
+namespace StepChoosers {
+template <typename StepChooserRegistrars>
+class RelaxTimeStepper;
+
+namespace Registrars {
+using RelaxTimeStepper =
+    Registration::Registrar<StepChoosers::RelaxTimeStepper>;
+}  // namespace Registrars
+
+/// Avoids instabilities due to rapid increases in the step size by
```

Comment:
I'm not sure I understand what exactly what is being done by this StepChooser (but that may be because I am tired...) so I can't suggest a better name...  Might be easier to discuss in person tomorrow...